Zend/zend_vm_def.h

```diff
@@ -7282,6 +7248,24 @@ ZEND_VM_HANDLER(149, ZEND_HANDLE_EXCEPTION, ANY, ANY)
 
 	cleanup_unfinished_calls(execute_data, throw_op_num);
 
+	if (throw_op->result_type & (IS_VAR | IS_TMP_VAR)) {
+		switch (throw_op->opcode) { 
+			case ZEND_ADD_ARRAY_ELEMENT:
+			case ZEND_ROPE_ADD:
+				break; /* exception while building structures, live range handling will free those */
+
+			case ZEND_FETCH_CLASS:
+			case ZEND_DECLARE_CLASS:
+			case ZEND_DECLARE_INHERITED_CLASS:
+			case ZEND_DECLARE_ANON_CLASS:
+			case ZEND_DECLARE_ANON_INHERITED_CLASS:
+				break; /* return value is zend_class_entry pointer */
+
+			default:
+				zval_ptr_dtor_nogc(EX_VAR(throw_op->result.var));
+		}
+	}
+
```

Comment:
You fixed add_function, but not sub_function and bitwise_not

$ USE_ZEND_ALLOC=0 valgrind sapi/cli/php -r '$b=[]; $b-2;'

==24197== Conditional jump or move depends on uninitialised value(s)
==24197==    at 0x8661711: _zval_ptr_dtor_nogc (zend_variables.h:39)
==24197==    by 0x866BC87: ZEND_HANDLE_EXCEPTION_SPEC_HANDLER (zend_vm_execute.h:1789)
==24197==    by 0x8668664: execute_ex (zend_vm_execute.h:429)